JumpRoyale/src/TimerOverlay.cs

```diff
@@ -0,0 +1,62 @@
+using System.Threading.Tasks;
+using Godot;
+
+public partial class TimerOverlay : VFlowContainer
+{
+    private const string SpriteName = "Sprite2D";
+    private const string TimerNodeName = SpriteName + "/Timer";
+
+    private static readonly int GameLength = GameOverlay.GameLength;
+
+    private int _timerSeconds = GameLength;
+
+    [Signal]
+    public delegate void TimerDoneEventHandler();
+
+    public void Init()
+    {
+        _ = StartTimer();
+    }
+
+    public async Task StartTimer()
+    {
+        await ToSignal(GetTree().CreateTimer(1.0f), SceneTreeTimer.SignalName.Timeout);
+
+        UpdateTimer();
+
+        if (_timerSeconds <= 0)
+        {
+            EmitSignal(SignalName.TimerDone);
+        }
+        else
+        {
+            _ = StartTimer();
+        }
+    }
+
+    private void UpdateTimer()
+    {
+        _timerSeconds--;
+
+        int seconds = _timerSeconds % 60;
+        int minutes = _timerSeconds / 60;
+
+        GetNode<Label>(TimerNodeName).Text = $"{minutes}:{seconds.ToString("00")}";
+        Sprite2D sprite = GetNode<Sprite2D>(SpriteName);
+        if (_timerSeconds > 15)
+        {
+            sprite.Modulate = new Color(1, 1, 1, 1);
+        }
+        else
+        {
+            if (seconds % 2 == 0)
+            {
+                sprite.Modulate = new Color(1, 1, 1, 1);
+            }
+            else
+            {
+                sprite.Modulate = new Color(1, 0, 0, 1);
+            }
+        }
```

Comment:
We can make this one shorter by eliminating nested condition with an early return, extracting its `else` block into a short conditional (assuming the colors were extracted).

```suggestion
        if (_timerSeconds > 15)
        {
            sprite.Modulate = _white;
            return;
        }

        sprite.Modulate = seconds % 2 == 0 ? _white : _red;
```

This could be even shorter if we store the `Sprite2D` in the field, initialize and modulate in `Init()`, keeping the above `if` with just return. Maybe I will play around on another refactoring round 😄